### PR TITLE
[MAINTENANCE] Automatic linting with pre-commit hook configuration

### PR DESCRIPTION
**Pull request by jcampbell**

### CODE REVIEW

**Comment by Aylr** [comment]
This is a day I've been waiting for for a long time. It brings joy to my 🖤 .

**Comment by abegong** [comment]
🎊 🎊 🎊 🎊 🎊

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 19.10b0
+    hooks:
+    -   id: black
+- repo: https://github.com/timothycrosley/isort
+  rev: 4.3.21-2
+  hooks:
+      - id: isort
```
Comment (mzjp2) [inline .pre-commit-config.yaml:16]: Might want to install a plugin for your editor to automatically include new lines at the end of file if they don't exist. There's actually a (built-in) pre-commit hook that does this for you 😉  see below for an example

https://github.com/quantumblacklabs/kedro/blob/9f95ee680012f7a20c31fcc820b7ed3a894ae9e6/.pre-commit-config.yaml#L11

File: requirements-dev.txt [hunk 2]
```diff
@@ -6,16 +6,15 @@ pytz>=2015.6
 jsonschema>=2.5.1
 altair>=3.1.0
 ruamel.yaml>=0.15.24
-ipywidgets>=7.4.2
+ipywidgets>=7.5.1
```
Comment (mzjp2) [inline requirements-dev.txt]: One other pre-commit hook you might want to install is `requirements-txt-fixer` -- automatically sorts your requirements, it's minor but quite nice 😀 -- see below for an example

https://github.com/quantumblacklabs/kedro/blob/9f95ee680012f7a20c31fcc820b7ed3a894ae9e6/.pre-commit-config.yaml#L20

**Review by mzjp2 — COMMENTED** [review]
I really like this. I've left a few comments and sorry for butting my head unasked into this PR 😅

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mzjp2) [inline .pre-commit-config.yaml]: I can appreciate the irony here 😅 

```suggestion
      - id: isort

```

File: requirements-dev.txt [hunk 3]
```diff
@@ -6,16 +6,15 @@ pytz>=2015.6
...
 requests>=2.20
-Click>=7.0
+Click>=7.1.2
```
Comment (mzjp2) [inline requirements-dev.txt]:
```suggestion
click>=7.1.2
```

consistency 😀

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jcampbell) [reply .pre-commit-config.yaml]: hahaahaha, love it! :)

File: requirements-dev.txt (hunk 2 above)
Comment (jcampbell) [reply requirements-dev.txt]: Added!

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    -   id: trailing-whitespace
+    - id: requirements-txt-fixer
+-   repo: https://github.com/psf/black
+    rev: 19.10b0
+    hooks:
+    -   id: black
+- repo: https://github.com/timothycrosley/isort
+  rev: 4.3.21-2
+  hooks:
+      - id: isort
```
Comment (Aylr) [inline .pre-commit-config.yaml]: I've found best results running `isort` first then `black`

**Pull request merged**